packages/block-library/src/social-link/edit.native.js

```diff
@@ -0,0 +1,188 @@
+const ANIMATION_DURATION = 400;
+
+const SocialLinkEdit = ( {
+	attributes,
+	setAttributes,
+	isSelected,
+	onFocus,
+} ) => {
+	const { url, service } = attributes;
+	const [ isLinkSheetVisible, setIsLinkSheetVisible ] = useState( false );
+	const [ isValue, setIsValue ] = useState( !! url );
+
+	const activeIcon =
+		styles[ `wp-social-link-${ service }` ] || styles[ `wp-social-link` ];
+
+	const inactiveIcon = usePreferredColorSchemeStyle(
+		styles.inactiveIcon,
+		styles.inactiveIconDark
+	);
+
+	const animatedValue = useRef( new Animated.Value( 0 ) ).current;
+
+	const IconComponent = getIconBySite( service )();
+	const socialLinkName = getNameBySite( service );
+
+	// When new social icon is added link sheet is opened automatically
+	useEffect( () => {
+		if ( isSelected && ! url ) {
+			setIsLinkSheetVisible( true );
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( ! url ) {
+			setIsValue( false );
+			animatedValue.setValue( 0 );
+		} else if ( url ) {
+			animateColors();
+		}
+	}, [ url ] );
+
+	const interpolationColors = {
+		backgroundColor: animatedValue.interpolate( {
+			inputRange: [ 0, 1 ],
+			outputRange: [
+				inactiveIcon.backgroundColor,
+				activeIcon.backgroundColor,
+			],
+		} ),
+		color: animatedValue.interpolate( {
+			inputRange: [ 0, 1 ],
+			outputRange: [ inactiveIcon.color, activeIcon.color ],
+		} ),
+		stroke: '',
+	};
+
+	const { backgroundColor, color, stroke } = isValue
+		? activeIcon
+		: interpolationColors;
+
+	const options = {
+		url: {
+			label: __( 'URL' ),
+			placeholder: __( 'Add URL' ),
+			autoFocus: true,
+		},
+		linkLabel: {
+			label: __( 'Link label' ),
+			placeholder: __( 'None' ),
+		},
+		footer: {
+			label: __( 'Briefly describe the link to help screen reader user' ),
+		},
+	};
+
+	function animateColors() {
+		Animated.sequence( [
+			Animated.delay( ANIMATION_DELAY ),
+			Animated.timing( animatedValue, {
+				toValue: 1,
+				duration: ANIMATION_DURATION,
+				easing: Easing.circle,
+			} ),
+		] ).start( () => setIsValue( true ) );
+	}
+
+	function onCloseSettingsSheet() {
+		setIsLinkSheetVisible( false );
+	}
+
+	function onEmptyURL() {
+		animatedValue.setValue( 0 );
+		setIsValue( false );
+	}
+
+	function onIconPress() {
+		if ( isSelected ) {
+			setIsLinkSheetVisible( true );
+		} else {
+			onFocus();
+		}
+	}
+
+	const accessibilityHint = url
+		? sprintf(
+				// translators: %s: social link name e.g: "Instagram".
+				__( '%s has URL set' ),
+				socialLinkName
+		  )
+		: sprintf(
+				// translators: %s: social link name e.g: "Instagram".
+				__( '%s has no URL set' ),
+				socialLinkName
+		  );
+
+	return (
+		<View>
+			{ isSelected && (
+				<BlockControls>
+					<ToolbarGroup>
+						<ToolbarButton
+							title={ sprintf(
+								// translators: %s: social link name e.g: "Instagram".
+								__( 'Add link to %s' ),
+								socialLinkName
+							) }
+							icon={ link }
+							onClick={ () => setIsLinkSheetVisible( true ) }
```

Comment:
I'm wondering why you created the `onCloseSettingsSheet` where you call `setIsLinkSheetVisible` with false but you did not create a function for this usage :)

Reply:
Looks like something which left after adjusting to new `LinkSettings` component, however, I will create an adequate function `onOpenSettingsSheet`.